addons/website/models/ir_attachment.py

```diff
@@ -26,11 +28,40 @@ def create(self, vals):
 
     @api.model
     def get_serving_groups(self):
-        return super(Attachment, self).get_serving_groups() + ['website.group_website_designer']
+        result = super(Attachment, self).get_serving_groups() + ['website.group_website_designer']
+        if request and request.endpoint.routing.get('routes')[0] == '/web_editor/modify_image/<model("ir.attachment"):attachment>':
+            # modify_image already performed specific read and write access checks
+            result = result + ['website.group_website_publisher']
+        return result
 
     @api.model
     def get_serve_attachment(self, url, extra_domain=None, extra_fields=None, order=None):
         website = self.env['website'].get_current_website()
         extra_domain = (extra_domain or []) + website.website_domain()
         order = ('website_id, %s' % order) if order else 'website_id'
         return super(Attachment, self).get_serve_attachment(url, extra_domain, extra_fields, order)
+
+    def _can_user_save_xml_image(self, user, values):
+        return super()._can_user_save_xml_image(user, values) or (
+            # White list restricted editor for modify_image route.
+            user.has_group('website.group_website_publisher') and (
+                (
+                    request and request.endpoint.routing.get('routes')[0] == '/web_editor/modify_image/<model("ir.attachment"):attachment>'
+                ) or self._is_known_xml_image(values)
+            )
+        )
+
+    def _is_known_xml_image(self, values):
+        """ Returns True if an XML is already known as an image
+            :param values: written values
+            :return: True if XML is already stored as XML image
+        """
+        if 'datas' in values and 'mimetype' in values:
+            bin_data = base64.b64decode(values['datas'] or b'')
+            checksum = self._compute_checksum(bin_data)
+            if self.search([
+                ('mimetype', '=', values['mimetype']),
+                ('checksum', '=', checksum),
+            ], limit=1):
```

Comment:
@bso-odoo I think we should ensure that res_model and res_id are the good one
```py
('res_model', '=', self.res_model),
('res_id', '=', self.res_id),
```
no ?

Reply:
At second thought, this introduces a lot of complexity:
E.g. in `website_sale` we will want to accept images from related `product.product`, `product.template` and `product.image`... computing this might be costly - and error prone.

AFAIK, the purpose of turning XML images into plain text is to avoid accepting malicious images. If a specific has been "whitelisted" because of the other conditions, I believe it is OK to still consider it whitelisted even if it is related to another record.